### PR TITLE
[env]Introduce flink-env test suite

### PR DESCRIPTION
**Pull request by ljz2051**
## What is the purpose of the change

This pull request introduces a test suite used for UI tests of flink-env interfaces

## Brief change log
- A flink-env test suite
- Mocked flink filesystem java class

## Verifying this change

This change added tests and can be verified by "FlinkEnvTest.java".

### CODE REVIEW

File: java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * by the Apache Software Foundation (ASF) under the Apache License, Version 2.0. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership.
+ */
+
+package org.apache.flink.core.fs;
+
+/**
+ * Interface that represents the client side information for a file independent of the file system.
+ */
+public interface FileStatus {
```
Comment (masteryhx) [inline java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java:30]: I am thinking maybe we could just let ForSt provide all these classes as a complete jar in the future.
Then some duplicate codes could be avoided.

File: java/src/main/java/org/rocksdb/EnvFlinkTestSuite.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.rocksdb;
+
+public class EnvFlinkTestSuite implements AutoCloseable {
```
Comment (masteryhx) [inline java/src/main/java/org/rocksdb/EnvFlinkTestSuite.java:26]: How about adding some comments, especially about how to add related test cases and use it ?

File: java/src/test/java/org/rocksdb/flink/FlinkEnvTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+package org.rocksdb.flink;
+
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.rocksdb.EnvFlinkTestSuite;
+import org.rocksdb.RocksNativeLibraryResource;
+
+public class FlinkEnvTest {
```
Comment (masteryhx) [inline java/src/test/java/org/rocksdb/flink/FlinkEnvTest.java:31]: nit: some comments

File: java/Makefile [hunk 4]
```diff
@@ -292,14 +295,15 @@ PLUGIN_SOURCES = $(foreach root, $(ROCKSDB_PLUGIN_JAVA_ROOTS), $(foreach pkg, or
...
+CORE_TEST_SOURCES = $(foreach pkg, org/rocksdb/test org/rocksdb/util org/rocksdb/flink org/rocksdb, $(TEST_SRC)/$(pkg)/*.java)
 TEST_SOURCES = $(wildcard $(CORE_TEST_SOURCES) $(PLUGIN_TEST_SOURCES))
+MOCK_FLINK_TEST_SOURCES = $(foreach pkg, org/apache/flink/core/fs  org/apache/flink/state/forst/fs, flinktestmock/src/main/java/$(pkg)/*.java)
 
 # Configure the plugin tests and java classes
 ROCKSDB_PLUGIN_NATIVE_JAVA_CLASSES = $(foreach plugin, $(ROCKSDB_PLUGINS), $(foreach class, $($(plugin)_NATIVE_JAVA_CLASSES), $(class)))
 NATIVE_JAVA_CLASSES = $(NATIVE_JAVA_CLASSES) $(ROCKSDB_PLUGIN_NATIVE_JAVA_CLASSES)
 ROCKSDB_PLUGIN_JAVA_TESTS = $(foreach plugin, $(ROCKSDB_PLUGINS), $(foreach testclass, $($(plugin)_JAVA_TESTS), $(testclass)))
-ALL_JAVA_TESTS = $(JAVA_TESTS) $(ROCKSDB_PLUGIN_JAVA_TESTS)
+ALL_JAVA_TESTS = $(FLINK_TESTS) $(JAVA_TESTS) $(ROCKSDB_PLUGIN_JAVA_TESTS)
```
Comment (masteryhx) [inline java/Makefile:306]: IIUC, current Github action doesn't include run_test, right ?

**Review by masteryhx — COMMENTED** [review]
Thanks for the PR.
PTAL my comments.

File: env/flink/env_flink.h [hunk 5]
```diff
@@ -115,6 +115,14 @@ class FlinkFileSystem : public FileSystemWrapper {
                          const IOOptions& /*options*/, IODebugContext* /*dbg*/,
                          jobject* /*fileStatus*/);
   std::string ConstructPath(const std::string& /*file_name*/);
+
+  static std::string TrimTrailingSlash(const std::string& base_path) {
```
Comment (fredia) [inline env/flink/env_flink.h:119]: nit: according to camel case naming, `trim` should be lowercase?

File: java/Makefile (hunk 4 above)
Comment (fredia) [reply java/Makefile:306]: `jtest` would be executed in `build-linux-java`  https://github.com/ververica/ForSt/blob/main/.github/workflows/pr-jobs.yml#L86

File: java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+ * limitations under the License.
+ */
+
+/*
+ * This file is based on source code from the Hadoop Project (http://hadoop.apache.org/), licensed
+ * by the Apache Software Foundation (ASF) under the Apache License, Version 2.0. See the NOTICE
+ * file distributed with this work for additional information regarding copyright ownership.
+ */
+
+package org.apache.flink.core.fs;
```
Comment (fredia) [inline java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java:25]: Will this result in potential package conflicts?

File: java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java (hunk 1 above)
Comment (ljz2051) [reply java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java:30]: Yes, I think we could provide flink-core and forst-statebackend jar packages for ForSt in the future.  At present, considering the problem of circular dependency, it needs to be bypassed in the current way.

File: java/src/main/java/org/rocksdb/EnvFlinkTestSuite.java (hunk 2 above)
Comment (ljz2051) [reply java/src/main/java/org/rocksdb/EnvFlinkTestSuite.java:26]: resolved.

File: java/Makefile (hunk 4 above)
Comment (ljz2051) [reply java/Makefile:306]: As fredia explained,  these UT tests can be executed by current github CI.

File: env/flink/env_flink.h (hunk 5 above)
Comment (ljz2051) [reply env/flink/env_flink.h:119]: For the other classes in ForSt, method names all start with uppercase letters,  so I think it's better to  stick to the project's code style?

File: java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java (hunk 6 above)
Comment (ljz2051) [reply java/flinktestmock/src/main/java/org/apache/flink/core/fs/FileStatus.java:25]: No. These files are placed in a separate package and are not included in javalib's SOURCES file[1], so they will not be compiled into Forst's javalib.

[1] https://github.com/ververica/ForSt/blob/ec88681c32e5f9d80b0bf331070bd05d740d685c/java/Makefile#L293C1-L293C56

**Review by masteryhx — APPROVED** [review]
Thanks for the update.
LGTM.

**Pull request merged**